bundles/org.openhab.binding.salus/src/main/java/org/openhab/binding/salus/internal/rest/JettyHttpClient.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.salus.internal.rest;
+
+import static java.util.Objects.requireNonNull;
+
+import javax.validation.constraints.NotNull;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.HttpResponseException;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.util.StringContentProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Martin Grześlowski - Initial contribution
+ */
+@NonNullByDefault
+public class JettyHttpClient implements RestClient {
+    private final Logger logger = LoggerFactory.getLogger(JettyHttpClient.class);
+    private final HttpClient client;
+
+    public JettyHttpClient(HttpClient client) {
+        this.client = requireNonNull(client, "client");
+        if (this.client.isStopped()) {
+            throw new IllegalStateException("HttpClient is stopped");
+        }
+    }
+
+    @Override
+    public Response<@Nullable String> get(String url, @Nullable Header... headers) {
+        var request = requireNonNull(client.newRequest(url));
+        return execute(request, headers, url);
+    }
+
+    @Override
+    public Response<@Nullable String> post(String url, @Nullable Content content, @Nullable Header... headers) {
+        var request = requireNonNull(client.POST(url));
+        if (content != null) {
+            request.content(new StringContentProvider(content.body()), content.type());
+        }
+        return execute(request, headers, url);
+    }
+
+    @NotNull
+    private Response<@Nullable String> execute(Request request, @Nullable Header[] headers, String url) {
+        try {
+            for (var header : headers) {
```

Comment:
I think headers==null to be handled, compiler warning

Reply:
I've fixed it but now I'm getting this warn:

```
[WARNING] D:\repos\openhab2-addons\bundles\org.openhab.binding.salus\src\main\java\org\openhab\binding\salus\internal\rest\JettyHttpClient.java:[59,17] Redundant null check: comparing 'RestClient.@Nullable Header @NonNull[]' against null
```

Reply:
Could you check if you mixed two different annotation frameworks? We use ```org.eclipse.jdt.annotation....```
```@NotNull``` is not a keyword we use....

Reply:
omg, sorry. 

The amount of null/nonnull annotations in Java ecosystem is killing me... Right now I can give you at least 4 packages....